src/firefly/js/ui/FileUploadDropdown.jsx

```diff
@@ -2,42 +2,54 @@
  * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
  */
 
-import React, {useState} from 'react';
+import React, {useContext, useState} from 'react';
 import {FormPanel} from './FormPanel.jsx';
 import {dispatchHideDropDown} from '../core/LayoutCntlr.js';
-import {panelKey, FileUploadViewPanel, resultSuccess, resultFail} from '../visualize/ui/FileUploadViewPanel.jsx';
+import {
+    FileUploadViewPanel,
+    resultFail
+} from '../visualize/ui/FileUploadViewPanel.jsx';
 import {getAppOptions} from 'firefly/api/ApiUtil.js';
 import DialogRootContainer from 'firefly/ui/DialogRootContainer.jsx';
 import {dispatchHideDialog, dispatchShowDialog} from 'firefly/core/ComponentCntlr.js';
 import {PopupPanel} from 'firefly/ui/PopupPanel.jsx';
+import {resultSuccess} from 'firefly/ui/FileUploadProcessor';
+import {FieldGroup} from 'firefly/ui/FieldGroup';
 
 const maskWrapper= { position:'absolute', left:0, top:0, width:'100%', height:'100%' };
+const panelKey = 'FileUploadAnalysis';
 
-export const FileUploadDropdown= ({style={}, onCancel=dispatchHideDropDown, onSubmit=resultSuccess}) =>{
+export const FileUploadDropdown= ({style={}, onCancel=dispatchHideDropDown, onSubmit=resultSuccess, acceptMoc}) =>{ //onSubmit=resultSuccess
     const [submitText,setSubmitText]= useState('Load');
     const [doMask, changeMasking]= useState(() => false);
     const helpId = getAppOptions()?.uploadPanelHelpId ?? 'basics.searching';
     return (
+
         <div style={{width: '100%', ...style}}>
-            <FormPanel
-                width='auto' height='auto' groupKey={panelKey} onSubmit={onSubmit}
-                onError={resultFail}
-                onCancel={onCancel}
-                submitText={submitText}
-                params={{hideOnInvalid: false}}
-                changeMasking={changeMasking}
-                inputStyle={{height:'100%'}}
-                submitBarStyle={{padding: '2px 3px 3px'}} help_id={helpId}>
-                <FileUploadViewPanel setSubmitText={setSubmitText}/>
-            </FormPanel>
+            <FieldGroup groupKey={panelKey} keepState={true} style={{height:'100%', width: '100%',
+                display: 'flex', alignItems: 'stretch', flexDirection: 'column'}}>
+                <FormPanel
+                    width='auto' height='auto' groupKey={panelKey} onSubmit={onSubmit}
+                    onError={resultFail}
+                    onCancel={onCancel}
+                    submitText={submitText}
+                    params={{hideOnInvalid: false}}
+                    changeMasking={changeMasking}
+                    inputStyle={{height:'100%'}}
+                    submitBarStyle={{padding: '2px 3px 3px'}} help_id={helpId}>
+                    <FileUploadViewPanel setSubmitText={setSubmitText} acceptMoc={acceptMoc}/>
```

Comment:
Thanks, good point. For now, the only reason setSubmitText is sent to `FileUploadViewPanel` is so that `FileUploadViewPanel` can dynamically change it by calling the getLoadButtonText() function to read something like "Load 3 images" (and then change that as more images/tables are selected/deselected). And getLoadButtonText() also makes use of some variables (tbl ID, report, summary model, etc.) that are local to `FileUploadViewPanel` and so that rules out being able to call getLoadButtonText() from `FileUploadDropdown`..